themes/default/content/docs/concepts/inputs-outputs/apply.md

```diff
@@ -0,0 +1,822 @@
+      toString: [Function (anonymous)],
+      toJSON: [Function (anonymous)]
+    }
+```
+
+{{% /choosable %}}
+
+{{% choosable language typescript %}}
+
+```bash
+# Example CLI output (truncated)
+Diagnostics:
+  pulumi:pulumi:Stack (aws-js-dev):
+    OutputImpl {
+      __pulumiOutput: true,
+      resources: [Function (anonymous)],
+      allResources: [Function (anonymous)],
+      isKnown: Promise { <pending> },
+      isSecret: Promise { <pending> },
+      promise: [Function (anonymous)],
+      toString: [Function (anonymous)],
+      toJSON: [Function (anonymous)]
+    }
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```bash
+# Example CLI output (truncated)
+Diagnostics:
+  pulumi:pulumi:Stack (aws-iac-dev):
+    Calling __str__ on an Output[T] is not supported.
+    To get the value of an Output[T] as an Output[str] consider:
+    1. o.apply(lambda v: f"prefix{v}suffix")
+    See https://www.pulumi.com/docs/concepts/inputs-outputs for more details.
+    This function may throw in a future version of Pulumi.
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```bash
+# Example CLI output (truncated)
+Diagnostics:
+  pulumi:pulumi:Stack (aws-go-dev):
+    {0xc000137180}
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```bash
+# Example CLI output (truncated)
+Diagnostics:
+  pulumi:pulumi:Stack (aws-csharp-dev):
+    Calling [ToString] on an [Output<T>] is not supported.
+    To get the value of an Output<T> as an Output<string> consider:
+    1. o.Apply(v => $"prefix{v}suffix")
+    2. Output.Format($"prefix{hostname}suffix");
+    See https://www.pulumi.com/docs/concepts/inputs-outputs for more details.
+    This function may throw in a future version of Pulumi.
+```
+
+{{% /choosable %}}
+
+{{% choosable language java %}}
+
+```bash
+# Example CLI output (truncated)
+Updating (pulumi/dev)
+    Type                                          Name           Status              Info
+ +   pulumi:pulumi:Stack                           aws-java-dev     created (1s)        391 messages
+ +   └─ awsx:ec2:Vpc                               vpc            created (1s)
+...
+...
+
+# Nothing is printed
+
+Resources:
+    + 34 created
+
+Duration: 2m17s
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+This example is not applicable in YAML.
+```
+
+{{% /choosable %}}
+
+This is where {{< pulumi-apply >}} comes into play. As mentioned before, all properties of a resource are of type Output<T>, meaning the values only become known after the infrastructure has been provisioned. When a Pulumi program is executed with `pulumi up`, the {{< pulumi-apply >}} function will wait for the resource to be created and for its properties to be resolved before printing the desired value of the property. This is not something a standard `print | log` statement is capable of doing.
+
+### Accessing single output values
+
+The syntax of {{< pulumi-apply >}} is shown below:
```

Comment:
HUGE +1 for adding language-specific syntax reference for `apply`. This was a big miss in our docs previously---great work!